### PR TITLE
SysGCM: Split default command buffer into 32kb big fragments and swap between them

### PR DESCRIPTION
**Pull request by vlj**
This commit fixes dynamic tests from rpcs3 autotest repo. It doesn't break Arkedo 2, Voodoo Chronicles, Sonic CD, the guided fate paradox, puzzle quest ; it doesn't fix them either, although flickering is a lot less present in Voodoo Chronicles and puzzle quest loading screen.

Unfortunatly it looks like there is more PPU Thread failure when dealing with syscall and I have no idea why. Not sure if it's a bug of the commit or if it uncovered one.

### CODE REVIEW

File: rpcs3/Emu/SysCalls/Modules/cellGcmSys.cpp [hunk 1]
```diff
@@ -1158,7 +1174,54 @@ s32 cellGcmCallback(vm::ptr<CellGcmContextData> context, u32 count)
...
+
+		std::this_thread::yield();
+	}
+
+	u32 address = contexts[currentContext].begin;
+	const u32 upper = offsetTable.ioAddress[address >> 20]; // 12 bits
+	assert(upper != 0xFFFF);
+	const u32 offset = (upper << 20) | (address & 0xFFFFF);
+
+	// TODO : it should be context->current but for some reasons it's not working
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/Modules/cellGcmSys.cpp]: I checkec original gcm callback functionality. It must use `current`, but writing `ctrl.put` must be the very first operation of `cellGcmCallback()`.

**Comment by Nekotekina** [comment]
Callback for CELL_GCM_DEFAULT_FIFO_MODE_TRADITIONAL mode:

1) Update `ctrl.put` with `context.current`
2) Calculate bounds of the next command buffer
3) Write jump command, update `context`
4) Wait on `ctrl.get`

I'll try to get more details later, but they may become obvious during implementation of this algorithm.

**Comment by vlj** [comment]
I updated the PR with your suggestion (no vector, reordering of operation).
Looks like there is no more PPU thread failure or crash.

**Comment by vlj** [comment]
As recommanded by Nekoteki I also check if get pointer is outside of command buffer as a whole (ie in a call/return construct)

**Comment by Nekotekina** [comment]
Seems OK for me. I'd also check `Emu.IsStopped()` but it's not important.

**Pull request merged**